pkg/network/dns_snooper.go

```diff
@@ -188,7 +172,14 @@ func (s *SocketFilterSnooper) processPacket(data []byte, ts time.Time) {
 
 func (s *SocketFilterSnooper) pollPackets() {
 	for {
-		data, captureInfo, err := s.source.ZeroCopyReadPacketData()
+		err := s.source.VisitPacket(func(data []byte, timestamp time.Time) error {
```

Comment:
So this seems to change the flow a little bit. The older code was checking for `s.exit` after calling `ZeroCopyReadPacketData`, but the newer code does not do that. In cases where we get an `EAGAIN`, wouldn't we want to bail if `s.exit` is closed, and not retry?

Reply:
Nice catch. Since we're calling VisitPackets in a loop, I should actually be able to remove the EGAIN logic altogether.